modules/components/src/DataTable/TableToolbar/helpers.js

```diff
@@ -0,0 +1,67 @@
+import download from '../../utils/download';
+
+export const saveTSV = async ({ url, files = [], fileName, options = {} }) =>
+  download({
+    url,
+    method: 'POST',
+    ...options,
+    params: {
+      fileName,
+      files: files.map(({ allColumns, columns, exColumns = null, ...file }, i) => ({
+        ...file,
+        columns: exColumns // if the component gave you custom columns to show
+          ? Object.values(
+              exColumns.length > 0 // if they ask for any specific columns
+                ? exColumns.map((fieldName) => allColumns[fieldName]) // use them
+                : allColumns, // else, they're asking for all the columns
```

Comment:
For future reference, this block is where the default columns are determined, if the client app gives us an empty array (which is not 'nothing'), in case we decide responding with all the fields in the index doesn't work.

Line 16, particularly, could instead have columns, to provide only the 'active' columns (i.e. as set in admin-ui) .